pkg/controllers/securitypolicy_associated_controller_test.go

```diff
@@ -0,0 +1,570 @@
+	pod2 := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-2",
+			Namespace: "kube-system",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{
+						{Name: "test-port", ContainerPort: 8080},
+					},
+				},
+			},
+		},
+	}
+	pod3 := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-3",
+			Namespace: "test",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{},
+				},
+			},
+		},
+	}
+	type args struct {
+		pod *v1.Pod
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{"1", args{&pod}, true},
+		{"2", args{&pod2}, false},
+		{"3", args{&pod3}, false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, checkPodHasNamedPort(*tt.args.pod, ""), "checkPodHasNamedPort(%v)", tt.args.pod)
+		})
+	}
+}
+
+func Test_getAllPodPortNames(t *testing.T) {
+	pod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "test-namespace",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{
+						{Name: "test-port", ContainerPort: 8080},
+						{Name: "test-port-2", ContainerPort: 80},
+					},
+				},
+			},
+		},
+	}
+	type args struct {
+		pods []v1.Pod
+	}
+	tests := []struct {
+		name string
+		args args
+		want sets.String
+	}{
+		{"1", args{[]v1.Pod{pod}}, sets.NewString("test-port", "test-port-2")},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, getAllPodPortNames(tt.args.pods), "getAllPodPortNames(%v)", tt.args.pods)
+		})
+	}
+}
+
+func TestEnqueueRequestForNamespace_Create(t *testing.T) {
+	type fields struct {
+		Client client.Client
+	}
+	type args struct {
+		createEvent event.CreateEvent
+		l           workqueue.RateLimitingInterface
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{"1", fields{}, args{event.CreateEvent{}, nil}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &EnqueueRequestForNamespace{
+				Client: tt.fields.Client,
+			}
+			e.Create(tt.args.createEvent, tt.args.l)
+		})
+	}
+}
+
+func TestEnqueueRequestForPod_Raw1(t *testing.T) {
+	evt := event.CreateEvent{
+		Object: &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-pod",
+			},
+		},
+	}
+	type fields struct {
+		Client client.Client
+	}
+	type args struct {
+		evt interface{}
+		q   workqueue.RateLimitingInterface
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{"1", fields{}, args{evt, nil}},
+	}
+	patches := gomonkey.ApplyFunc(reconcileSecurityPolicy, func(client client.Client, pods []v1.Pod,
+		q workqueue.RateLimitingInterface,
+	) error {
+		return nil
+	})
+	defer patches.Reset()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &EnqueueRequestForPod{}
+			e.Raw(tt.args.evt, tt.args.q)
+		})
+	}
```

Comment:
It seems that you only mock the `reconcileSecurityPolicy` and then invoke `Raw`, but didn't verify anything.

Reply:
Yes, because the Raw function returns no error, this test mainly ensures that it supports different events.